indico/modules/categories/forms.py

```diff
@@ -65,12 +68,40 @@ class CategorySettingsForm(IndicoForm):
     notify_managers = BooleanField(_('Notify managers'), widget=SwitchWidget(),
                                    description=_('Whether to send email notifications to all managers of this category '
                                                  'when an event is created inside it or in any of its subcategories.'))
+    google_wallet_enabled = BooleanField(_('Google Wallet'), widget=SwitchWidget(),
+                                         description=_('Add Google Wallet configuration.'))
+    google_wallet_application_credentials = TextAreaField(_('Google Wallet application credentials'),
+                                                          [DataRequired(),
+                                                           HiddenUnless('google_wallet_enabled', preserve_data=True)],
+                                                          description=_('Json content generated from Google Wallet '
+                                                                        'console.'))
+    google_wallet_issuer_name = StringField(_('Google Wallet Issuer Name'),
+                                            [DataRequired(),
+                                             HiddenUnless('google_wallet_enabled', preserve_data=True)],
+                                            description=_('Issuer name that will appear in the Google Wallet ticket '
+                                                          'top header.'))
+    google_wallet_issuer_id = StringField(_('Google Wallet Issuer ID'),
+                                          [DataRequired(),
+                                           HiddenUnless('google_wallet_enabled', preserve_data=True)],
+                                          description=_('Issuer ID assigned in the Google Wallet console.'))
+
     event_creation_notification_emails = EmailListField(_('Notification E-mails'),
                                                         description=_('List of emails that will receive a notification '
                                                                       'every time a new event is created inside the '
                                                                       'category or one of its subcategories. '
                                                                       'One email address per line.'))
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not config.ENABLE_GOOGLE_WALLET:
+            new_basic_fields = []
+            for field in self.BASIC_FIELDS:
+                if not field.startswith('google_wallet'):
+                    new_basic_fields.append(field)
+                else:
+                    delattr(self, field)
+            self.BASIC_FIELDS = tuple(new_basic_fields)
```

Comment:
Is this still necessary? I think as long as you delete all the fields (which you do now), it should work without changing `BASIC_FIELDS`

Reply:
Tested, you are right.
I just have to add an additional check for the form submit.
Done.